src/sequential/sequential.hpp

```diff
@@ -0,0 +1,198 @@
+#pragma once
+
+#include <cassert>
+#include <cstdint>
+#include <iostream>
+#include <utility>
+#include <vector>
+
+
+////////////////////////////////////////////////////////////////////////////////
+/// TYPE DEFINITIONS (for easy refactor)
+////////////////////////////////////////////////////////////////////////////////
+
+/// N.B.  I use int32_t for the key and value so that they fit into a 64-bit word
+///       if I so choose to do that. This way, we can do atomic updates. The
+///       extra metadata is a bit of an oof though.
+/// N.B.  I do not use *_t because this is a reserved name in POSIX.
+/// N.B.  I also named them something unique so it's easy to find/replace.
+typedef int32_t KeyType;
```

Comment:
I think I see why you used `reinterpret_cast` now. Is there any reason `KeyType` can't be `uint32_t`?

Reply:
Nope but why would we use uint32_t? Oh because the conversion from `int32_t` to `size_t`?

Reply:
That's part of it, but also because I'd rather not deal with possible undefined behaviour with signed numbers if we don't have to.